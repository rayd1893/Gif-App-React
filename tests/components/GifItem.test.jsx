const { render, screen } = require("@testing-library/react")
const { GifItem } = require("../../src/components/GifItem")

describe('Pruebas en <GifItem />', () => { 
    const title = 'Saitama';
    const url = 'https://one-punch.com/saitama.jpg';
    test('It be match with snapshot', () => { 
        const { container } = render(<GifItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();
     });

     test('should show the image with the URL and ALT correctly', () => { 
        render(<GifItem title={ title } url={ url } />)
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe( url );
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
      });

      test('should show teh title', () => { 
        render(<GifItem title={ title } url={ url } />);
        expect(screen.getByText(title)).toBeTruthy();
       })
 })