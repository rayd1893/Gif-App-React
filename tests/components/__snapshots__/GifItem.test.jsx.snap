// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> It be match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
